### PR TITLE
feat: comments cleanup and license header file

### LINKED ISSUE #362: License header duplicated
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description

_please describe the error that you encountered_
As in a lot of projects I have a license header at the beginning of any file.
The result bundle contain all this license header replicated. There is a way to remove all the duplicated license header and keep only one at the beginning?

#### How To Reproduce

_please include instructions how to reproduce / for features: do you have a first draft or idea how to implement?_

#### Expected Behaviour

_please describe what behaviour or result you expected_

#### Version Information
last 2.0.0-rc.4
```
ng-packagr: v1.x.y
node: v8.x.y
@angular: v4.x.y
rxjs:
zone.js:
```

_please include any version information that might be relevant, e.g. other third-party libraries_

### PR DESCRIPTION
**Pull request by eromano**
## I'm submitting a...

```
[ ] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [x] Tests for the changes have been added

## Description

The change sets adds two new plugins when writing flat bundle files w/ rollup: 
 - "rollup-plugin-cleanup": "^2.0.0",
 - "rollup-plugin-license": "^0.5.0",

With this two new plugin and two new configurable property you can remove all the license header and optionally also other comments in the code and add only one license header on top of bundles.

All the previous behaviour are preserved those new plugins are activated only if configured. 
- the default behaviour of the license header plugin is disabled.
- the default behaviour of the comments plugin is 'all' that means keep all the comments

Closes #362 

#### Comments cleaning

To remove comments from the final bundles:

```
{
  "$schema": "../../../src/ng-package.schema.json",
  "lib": {
    "entryFile": "public_api.ts",
    "comments" : "none"
  }
}
```

The comments property can accept also a regex to selectively remove comments.

#### License header

To prepend a license header on your budles:

```
{
  "$schema": "../../../src/ng-package.schema.json",
  "lib": {
    "entryFile": "public_api.ts",
    "licensePath": "LICENSE"
  }
}
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by eromano** [comment]
Hi I just want add some context around it when you have a project with a license header in any file, you don't want it replicated inside your final bundle but you want it only once at the beginning.
With this functionality you can add your license header and delete with a regex all the license header duplicated.

**Comment by dherges** [comment]
Hi @eromano,

The intgration tests could be more minimal, but looks good to me!

Can you remove the package-lock.jsom from the PR?

**Comment by eromano** [comment]
Cool , yep I'll do it

**Comment by eromano** [comment]
@dherges done

File: package.json [hunk 1]
```diff
@@ -95,6 +95,8 @@
     "pretty-quick": "^1.2.1",
     "react": "^16.0.0",
     "react-dom": "^16.0.0",
+    "rollup-plugin-cleanup": "^2.0.0",
+    "rollup-plugin-license": "^0.5.0",
```
Comment (dherges) [inline package.json]: The plugins should go to the `dependencies` section.

**Pull request merged**